File: states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var OFF = {
  id: 1,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_OFF = 1;
var MOTIONMACHINE = {
  id: 2,
  master: {
    values: {
      
    },
    events: {
      goHome: function goHome() { manager.sendEvent(0, 2); },
      stairsUp: function stairsUp() { manager.sendEvent(1, 2); },
      stairsStop: function stairsStop() { manager.sendEvent(2, 2); },
      openDoor: function openDoor() { manager.sendEvent(3, 2); },
      closeDoor: function closeDoor() { manager.sendEvent(4, 2); },
      goUp: function goUp() { manager.sendEvent(5, 2); },
      goUpWithBall: function goUpWithBall() { manager.sendEvent(6, 2); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_MOTIONMACHINE = 2;

var STATES = {
  IDLE: IDLE,
  OFF: OFF,
  MOTIONMACHINE: MOTIONMACHINE
};
var manager = new Manager([IDLE, OFF, MOTIONMACHINE]);

File: MotionMachineAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[3] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {OFF::setup, OFF::enter, OFF::exit, OFF::loop, OFF::event},
  {MOTIONMACHINE::setup, MOTIONMACHINE::enter, MOTIONMACHINE::exit, MOTIONMACHINE::loop, MOTIONMACHINE::event}
};

static const WireValue wire_values[0] = {
  
};

MasterManager<State, 3, 0> manager(0x8e868773, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace OFF {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace MOTIONMACHINE {


void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::goHome();
    break;
  case 1:
    events::stairsUp();
    break;
  case 2:
    events::stairsStop();
    break;
  case 3:
    events::openDoor();
    break;
  case 4:
    events::closeDoor();
    break;
  case 5:
    events::goUp();
    break;
  case 6:
    events::goUpWithBall();
    break;
  default:
    break;
  }
}


}

